Duplicate package sources warning on macOS

**Describe the bug**
I pulled the project down to my M1 MBP to verify that things ran correctly. JetBrains Rider would not load dependencies and threw an error. `dotnet build` on the command line worked, but threw warnings for the three projects in `src/`:

`/Users/edmistond/projects/Smilodon/test/WebApp.Test/WebApp.Test.csproj : warning NU1507: There are 2 package sources defined in your configuration. When using central package management, please map your package sources with package source mapping (https://aka.ms/nuget-package-source-mapping) or specify a single package source. The following sources are defined: nuget.org, /usr/local/share/dotnet/library-packs [/Users/edmistond/projects/Smilodon/Smilodon.sln]`

**Expected behavior**
Project should build cleanly regardless of platform.

**Additional context**
I tested adding a nuget.config to my project root directory which simplify specifies nuget.org as the default wildcard package source, and this resolved the issue. I haven't opened the PR for it yet because I haven't had time to pull my fix branch down to my Windows desktop to ensure it doesn't break anything there. Wanted to make note of the issue and have something to file the PR against when I submit it.

Probably won't be able to get to this until Friday or Saturday. You can see my nuget.config here: https://github.com/edmistond/Smilodon/commit/32eecbc30d3d5e364a6829b030cc498684728a6c

Good catch. This reminds me that we need to matrix the PR build as well.

I think we can define our package source to avoid this (probably what's in
your link).

On Wed, Nov 23, 2022, 11:30 PM David Edmiston ***@***.***>
wrote:

> *Describe the bug*
> I pulled the project down to my M1 MBP to verify that things ran
> correctly. JetBrains Rider would not load dependencies and threw an error. dotnet
> build on the command line worked, but threw warnings for the three
> projects in src/:
>
> /Users/edmistond/projects/Smilodon/test/WebApp.Test/WebApp.Test.csproj :
> warning NU1507: There are 2 package sources defined in your configuration.
> When using central package management, please map your package sources with
> package source mapping (https://aka.ms/nuget-package-source-mapping) or
> specify a single package source. The following sources are defined:
> nuget.org, /usr/local/share/dotnet/library-packs
> [/Users/edmistond/projects/Smilodon/Smilodon.sln]
>
> *Expected behavior*
> Project should build cleanly regardless of platform.
>
> *Additional context*
> I tested adding a nuget.config to my project root directory which simplify
> specifies nuget.org as the default wildcard package source, and this
> resolved the issue. I haven't opened the PR for it yet because I haven't
> had time to pull my fix branch down to my Windows desktop to ensure it
> doesn't break anything there. Wanted to make note of the issue and have
> something to file the PR against when I submit it.
>
> Probably won't be able to get to this until Friday or Saturday. You can
> see my nuget.config here: ***@***.***
> <https://github.com/edmistond/Smilodon/commit/32eecbc30d3d5e364a6829b030cc498684728a6c>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DevChatter/Smilodon/issues/20>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARDXKTTTIIBFTPAFF2SNQLWJ3VQFANCNFSM6AAAAAASJWBBA4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

--- a/.gitignore
+++ b/.gitignore
@@ -348,3 +348,4 @@ MigrationBackup/
 
 # Ionide (cross platform F# VS Code tools) working folder
 .ionide/
+.idea/

--- a/.gitignore
+++ b/.gitignore
@@ -349,3 +349,6 @@ MigrationBackup/
 # Ionide (cross platform F# VS Code tools) working folder
 .ionide/
 .idea/
+
+# for now, ignore vscode configs
+.vscode/

--- a/nuget.config
+++ b/nuget.config
@@ -1,0 +1,17 @@
+<configuration>
+    <!-- `clear` ensures no additional sources are inherited from another config file. -->
+    <packageSources>
+        <clear />
+        <!-- `key` can be any identifier for your source. -->
+        <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
+    </packageSources>
+
+    <!-- Define mappings by adding package patterns beneath the target source. -->
+    <!-- Contoso.* packages and NuGet.Common will be restored from contoso.com, everything else from nuget.org. -->
+    <packageSourceMapping>
+        <!-- key value for <packageSource> should match key values from <packageSources> element -->
+        <packageSource key="nuget.org">
+            <package pattern="*" />
+        </packageSource>
+    </packageSourceMapping>
+</configuration>

